src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -701,10 +702,24 @@ impl<'tcx> Interval<'tcx> {
             ty::TyUint(_) | ty::TyChar => {
                 (lo, hi)
             }
-            _ => bug!("`Interval` should only contain integer types")
+            _ => bug!("`IntRange` should only contain integer types")
         }
     }
 
+    fn encode(tcx: TyCtxt<'_, 'tcx, 'tcx>,
+              ty: Ty<'tcx>,
+              range: RangeInclusive<u128>)
+              -> (u128, u128) {
+        Self::convert(tcx, ty, range, true)
+    }
+
+    fn decode(tcx: TyCtxt<'_, 'tcx, 'tcx>,
+              ty: Ty<'tcx>,
+              range: RangeInclusive<u128>)
+              -> (u128, u128) {
+        Self::convert(tcx, ty, range, false)
```

Comment:
It'd be nice to have two separate signatures (i.e. `encode` can produce `Self` and `decode` take `self`) and maybe not a `convert` method.